### PR TITLE
Upgrade to Django 2.2.x Python 3.9.x plus trivial changes

### PR DESCRIPTION
**Pull request by fpierfed**
This requires fixing the database to solve its own inconsistencies. I have a fixed DB and could generate a script to make the changes if needed.

### CODE REVIEW

File: requirements-dev.in [hunk 1]
```diff
@@ -6,7 +6,8 @@ pytest-xdist
 
 freezegun
 responses
-factory_boy
+# factory_boy
+git+https://github.com/FactoryBoy/factory_boy.git#factory_boy
```
Comment (matrixise) [inline requirements-dev.in:10]: why the git repository?

File: pycon/settings.py [hunk 2]
```diff
@@ -212,7 +212,6 @@ def _(x):
 
 MIDDLEWARE = (
     'django.contrib.sessions.middleware.SessionMiddleware',
-    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
```
Comment (matrixise) [inline pycon/settings.py:215]: why this remove?

Comment (fpierfed) [reply pycon/settings.py:215]: It is not needed anymore with Django 2.0 and later: https://docs.djangoproject.com/en/3.1/releases/2.0/

File: requirements-dev.in (hunk 1 above)
Comment (fpierfed) [reply requirements-dev.in:10]: There is crash when locale is not specified otherwise. It is fixed in master but not in the released version yet.

**Review by matrixise — APPROVED** [review]

**Pull request merged**